src/sentry/api/endpoints/artifact_bundles.py

```diff
@@ -0,0 +1,110 @@
+        :pparam string project_slug: the slug of the project to list the
+                                     artifact bundles of.
+        """
+        try:
+            queryset = ProjectArtifactBundle.objects.filter(
+                organization_id=project.organization_id, project_id=project.id
+            ).values("artifact_bundle_id", "date_added")
+        except ProjectArtifactBundle.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        def expose_artifact_bundle(artifact_bundle, artifact_bundle_meta):
+            bundle_id, artifact_count = artifact_bundle_meta
+
+            return {
+                "type": "artifact_bundle",
+                "bundleId": bundle_id,
+                "date": artifact_bundle["date_added"],
+                "fileCount": artifact_count,
+            }
+
+        def serialize_results(results):
+            artifact_bundle_counts = ArtifactBundle.get_artifact_counts(
+                [r["artifact_bundle_id"] for r in results]
+            )
+
+            # We want to maintain the -date_added ordering, thus we index the metadata by using the id fetched with the
+            # first query.
+            return serialize(
+                [
+                    expose_artifact_bundle(
+                        artifact_bundle=r,
+                        artifact_bundle_meta=artifact_bundle_counts[r["artifact_bundle_id"]],
+                    )
+                    for r in results
+                    if r["artifact_bundle_id"] in artifact_bundle_counts
+                ],
+                request.user,
+            )
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by="-date_added",
+            paginator_cls=OffsetPaginator,
+            default_per_page=10,
+            on_results=serialize_results,
+        )
+
+    def delete(self, request: Request, project) -> Response:
+        """
+        Delete an Archive
+        ```````````````````````````````````````````````````
+
+        Delete all artifacts inside given archive.
+
+        :pparam string organization_slug: the slug of the organization the
+                                            archive belongs to.
+        :pparam string project_slug: the slug of the project to delete the
+                                        archive of.
+        :qparam string name: The name of the archive to delete.
+        :auth: required
+        """
+
+        bundle_id = request.GET.get("bundle_id")
+
+        if bundle_id:
+            with atomic_transaction(using=router.db_for_write(ArtifactBundle)):
+                try:
+                    artifact_bundle = ArtifactBundle.objects.get(
+                        organization_id=project.organization_id, bundle_id=bundle_id
+                    )
+                    # We want to delete all the connections to a project.
+                    ProjectArtifactBundle.objects.filter(
```

Comment:
I think here: you want to set atomic transactions only on mutating queries, rather than for SELECT. So I would rewrite it:
1. Check bundle
2. query it:
```
try:

  artifact_bundle = ArtifactBundle.objects.get(
                          organization_id=project.organization_id, bundle_id=bundle_id
                      )
except ArtifactBundle.DoesNotExist:
    return Response(status=404)
    ^^^ may be be more informative that `bundle not found` and status code 404
```

and then start transaction to delete from 2 tables or rollback.:
```
with atomic_transaction(using=router.db_for_write(ArtifactBundle)):
```